python/tank_vendor/shotgun_authentication/interactive_authentication.py

```diff
@@ -330,50 +349,59 @@ def _process_ui():
 
 # Lock the assures only one thread at a time can execute the authentication logic.
 _renew_session_internal_lock = threading.Lock()
```

Comment:
again, wondering if this is worth wrapping up in a class just to give a bit of scope. Maybe even as part of `UiAuthenticationHandler`?

Then your `AuthState` class could just be a set of constants. In the rest of toolkit, we have adopted this convention for writing that sort of structure:

```
class UiAuthenticationHandler(object):

    AUTH_WAITING, AUTH_SUCCESS, AUTH_CANCELLED = range(3)

```